src/SearchResults.js

```diff
@@ -0,0 +1,164 @@
+    if (
+      sortConfig &&
+      sortConfig.key === key &&
+      sortConfig.direction === "ascending"
+    ) {
+      direction = "descending";
+    }
+    setSortConfig({ key, direction });
+  };
+
+  return { items: sortedItems, getSorted, sortConfig };
+};
+
+const SearchResults = props => {
+  const [customerId, setCustomerId] = useState("");
+  const [color, setColor] = useState([]); // set color == empty array
+
+  const { items, getSorted } = SortableData(props.BookingsResults);
+
+  const getCustomerId = id => {
+    setCustomerId(id);
+  };
+
+  const highlightRow = currentIndex => {
+    if (!color.includes(currentIndex)) {
+      setColor([...color, currentIndex]);
+    } else {
+      setColor(color.filter(item => item !== currentIndex));
+    }
+  };
+
+  //className="table-container"
+
+  return (
+    <div>
+      <table className="table  results">
+        <thead>
+          <tr>
+            <th scope="col">Id</th>
+            <th scope="col">Title</th>
+            <th
+              className="pointer"
+              onClick={() => getSorted("firstName")}
+              scope="col"
+            >
+              First name
+            </th>
+            <th
+              className="pointer"
+              onClick={() => getSorted("surname")}
+              scope="col"
+            >
+              Surname
+            </th>
+            <th
+              className="pointer"
+              onClick={() => getSorted("email")}
+              scope="col"
+            >
+              Email
+            </th>
+            <th
+              className="pointer"
+              onClick={() => getSorted("roomId")}
+              scope="col"
+            >
+              Room Id
+            </th>
+            <th
+              className="pointer"
+              onClick={() => getSorted("checkInDate")}
+              scope="col"
+            >
+              Check in date
+            </th>
+            <th
+              className="pointer"
+              onClick={() => getSorted("checkOutDate")}
+              scope="col"
+            >
+              Check out date
+            </th>
+            <th scope="col">Number of Night</th>
+          </tr>
+        </thead>
+        <tbody>
+          {items.map((item, currentRowIndex) => {
+            const {
+              id,
+              title,
+              firstName,
+              surname,
+              email,
+              roomId,
+              checkInDate,
+              checkOutDate
+            } = item;
+
+            const startDate = moment(item.checkInDate);
+            const endDate = moment(item.checkOutDate);
+            let numberOfNights = endDate.diff(startDate, "days", true);
+
+            return (
+              <tr
+                style={
+                  color.includes(currentRowIndex)
+                    ? { backgroundColor: "orange" }
+                    : { backgroundColor: "transparent" }
+                }
```

Comment:
This is an interesting use of conditional formatting, and here it works well, to make it even better maybe consider applying style classes conditionally, instead of inline styling. This sort of change can make your code easier to maintain in a professional setting.